python/inc/bond/python/converters.h

```diff
@@ -181,12 +181,16 @@ extend_map(Map& map, const boost::python::object& obj)
     typedef typename Map::value_type value_type;
 
     BOOST_ASSERT(PyDict_Check(obj.ptr()));
-    
+
     dict dict(obj);
-        
+
     BOOST_FOREACH(object k,
         std::make_pair(
+#if PY_VERSION_HEX >= 0x03000000
```

Comment:
This doesn't compile under Visual Studio 2013. `BOOST_FOREACH` is a macro and using preprocessor directives within macro arguments has an undefined behavior.

I'd suggest moving the conditional code outside of `BOOST_FOREACH`:

~~~
const auto keys = 
#if PY_VERSION_HEX >= 0x03000000
            dict.keys();
#else
            dict.iterkeys();
#endif 
~~~